include/sparrow_ipc/serialize_primitive_array.hpp

```diff
@@ -50,18 +53,22 @@ namespace sparrow_ipc
     }
 
     template <typename T>
-    sparrow::primitive_array<T> deserialize_primitive_array(const std::vector<uint8_t>& buffer) {
+    sparrow::primitive_array<T> deserialize_primitive_array(const std::vector<uint8_t>& buffer)
+    {
         const uint8_t* buf_ptr = buffer.data();
         size_t current_offset = 0;
 
         // I - Deserialize the Schema message
         std::optional<std::string> name;
         std::optional<std::vector<sparrow::metadata_pair>> metadata;
-        details::deserialize_schema_message(buf_ptr, current_offset, name, metadata);
+        deserialize_schema_message(std::span<const uint8_t>(buffer), current_offset, name, metadata);
 
         // II - Deserialize the RecordBatch message
```

Comment:
I'm a bit surprised by the logic here. Typically, an IPC stream has a single Schema message at start, followed by an arbitrary number of RecordBatch messages (optionally other messages as well). Here you seem to be assuming that the IPC stream will only ever contain a single RecordBatch, which is weird.

Reply:
In other words, I would not expect a single stateless function to handle both Schema and RecordBatch at once.

I'm also surprised that this function knows the desired type `primitive_array<T>` *before* deserializing the Schema.

Reply:
I only reworked the deserialization. The serialization will be in another PR. Here it's the old implementation which is kept for tests. But it will be removed soon.